libs/math/include/math/activation_functions/softmax.hpp

```diff
@@ -99,9 +99,20 @@ void Softmax(ArrayType const &array, ArrayType &ret, typename ArrayType::SizeTyp
   {
     details::Softmax2DImplementation(array, ret, axis);
   }
+  else if ((array.shape().size() == 3) && (ret.shape().size() == 3) && (axis == 0 || axis == 1))
+  {
+    ArrayType tmp_ret = ret.Slice(0, 2).Copy().Squeeze();
```

Comment:
this is sort of redundant:
what you really want to do here is something like this:
ArrayType buffer({array.shape(0}, array.shape(1)});